### PR TITLE
Adapt QQBar documentation

### PR DESCRIPTION
**Pull request by lgoettgens**
Some more progress towards https://github.com/oscar-system/Oscar.jl/issues/3416.

This follows the approach of https://github.com/oscar-system/Oscar.jl/issues/3416#issuecomment-2047092266, i.e.:

- preparing for removal of `CalciumQQBar` (global constant)
- preparing for unexporting `QQBar` (global constant)
- adapt the documentation and tests to use `algebraic_closure(QQ)` instead.

In the next breaking release of Nemo, a few things marked with a `TODO` can be removed. This will only be a breaking change for Nemo, but not for Oscar.

### CODE REVIEW

**Comment by lgoettgens** [comment]
> Test failure looks real

Some copy-and-paste error when adapting the tests probably. But needs to wait for me to get back to a computer

File: docs/src/algebraic.md [hunk 1]
```diff
@@ -132,7 +120,7 @@ julia> CC = AcbField(32); CC(QQBar(-1) ^ (QQBar(1) // 4))
 Retrieving the minimal polynomial and algebraic conjugates
 of a given algebraic number:
 
-```jldoctest
+```jldoctest; setup = :(QQBar = algebraic_closure(QQ))
```
[the remark this answers is not in the document]
Comment (lgoettgens) [reply docs/src/algebraic.md:123]: Just wanting to mention here that the same reasoning should be applied to all of the setup blocks where we create `RR = RealField()` and `CC = ComplexField()` etc.

File: test/calcium/qqbar-test.jl [hunk 2]
```diff
@@ -35,38 +35,39 @@
...
-
-   @test iszero(R(0))
-   @test isone(R(1))
-   @test is_rational(R(1))
-   @test isreal(R(1))
-   @test degree(R(1)) == 1
-
-   u = sqrt(R(2))
-   i = sqrt(R(-1))
+   QQBar = algebraic_closure(QQ)
```
Comment (fingolfin) [inline test/calcium/qqbar-test.jl:51]: If this was just
```suggestion
   R = algebraic_closure(QQ)
```
the diff would be much smaller

File: docs/src/algebraic.md [hunk 3]
```diff
@@ -260,7 +248,7 @@ is_less_root_order(a::QQBarFieldElem, b::QQBarFieldElem)
 
 **Examples**
 
-```jldoctest
+```jldoctest; setup = :(QQBar = algebraic_closure(QQ))
```
Comment (fingolfin) [inline docs/src/algebraic.md:251]: I guess we could just search&replace this to:
````suggestion
```jldoctest
julia> QQBar = algebraic_closure(QQ);

````

File: src/calcium/qqbar.jl [hunk 4]
```diff
@@ -10,15 +10,15 @@
 #
 ###############################################################################
 
-parent(a::QQBarFieldElem) = CalciumQQBar
+parent(a::QQBarFieldElem) = QQBar
```
Comment (fingolfin) [inline src/calcium/qqbar.jl:13]: Could just avoid the global var altogether, since it is a singleton it is equivalent to write
```suggestion
parent(a::QQBarFieldElem) = QQBarField()
```

File: src/calcium/qqbar.jl [hunk 5]
```diff
@@ -175,9 +175,9 @@ zero(a::QQBarField) = a(0)
 
 one(a::QQBarField) = a(1)
 
-zero(::Type{QQBarFieldElem}) = CalciumQQBar(0)
+zero(::Type{QQBarFieldElem}) = QQBar(0)
```
Comment (fingolfin) [inline src/calcium/qqbar.jl:178]:
```suggestion
zero(::Type{QQBarFieldElem}) = QQBarFieldElem(0)
```

File: src/calcium/qqbar.jl [hunk 6]
```diff
@@ -175,9 +175,9 @@ zero(a::QQBarField) = a(0)
...
 
-one(::Type{QQBarFieldElem}) = CalciumQQBar(1)
+one(::Type{QQBarFieldElem}) = QQBar(1)
```
Comment (fingolfin) [inline src/calcium/qqbar.jl:180]:
```suggestion
one(::Type{QQBarFieldElem}) = QQBarFieldElem(1)
```

File: test/calcium/qqbar-test.jl [hunk 7]
```diff
@@ -344,7 +343,7 @@ end
 
 
 @testset "QQBarFieldElem.inplace" begin
-   R = CalciumQQBar
+  R = algebraic_closure(QQ)
```
Comment (fingolfin) [inline test/calcium/qqbar-test.jl:346]:
```suggestion
   R = algebraic_closure(QQ)
```

**Comment by lgoettgens** [comment]
I think [`5c8674d` (#1715)](https://github.com/Nemocas/Nemo.jl/pull/1715/commits/5c8674df37ccd9e39aceaa39b2a36bc082b4d659) together with the sentence that was already there is what you imagined @fingolfin ?

File: docs/src/algebraic.md [hunk 8]
```diff
@@ -38,8 +28,7 @@ For fast calculation in $\overline{\mathbb{Q}}$,
 `CalciumField` should typically be used instead (see the section
 on *Exact real and complex numbers*).
 Alternatively, to compute in a fixed subfield of $\overline{\mathbb{Q}}$,
-you may fix a generator $a$ and construct an
-Antic number field to represent $\mathbb{Q}(a)$.
+you may fix a generator $a$ and construct a number field to represent $\mathbb{Q}(a)$.
```
Comment (fingolfin) [inline docs/src/algebraic.md:31]: I realize you are just editing existing text, so this is not a change request for this PR, but I genuinely wonder how to read this. Is `a` supposed to be an element of `CalciumField` ?! But in general were would `a` come from? Shouldn't we just say something like 

> Alternatively, one can often work in a suitable number field created via [`number_field`](@ref).

File: docs/src/algebraic.md [hunk 9]
```diff
@@ -10,23 +10,13 @@ end
...
-`CalciumQQBar`. Note that this is the name of a specific parent object,
-not the name of its type.
-
-
- Library        | Element type  | Parent type
-----------------|---------------|--------------------
-Calcium         | `QQBarFieldElem`       | `QQBarField`
+ Library        | Element type     | Parent type
+----------------|------------------|--------------------
+Calcium         | `QQBarFieldElem` | `QQBarField`
```
Comment (fingolfin) [inline docs/src/algebraic.md:19]: Again not part of this PR, but should we really still refer to "Calcium" here (and arb elsewhere), given that they have all been merged into FLINT?

File: test/calcium/qqbar-test.jl [hunk 10]
```diff
@@ -1,14 +1,14 @@
...
+  @test R == Nemo.QQBar
 
   @test elem_type(R) == QQBarFieldElem
   @test elem_type(QQBarField) == QQBarFieldElem
   @test parent_type(QQBarFieldElem) == QQBarField
   @test is_domain_type(QQBarFieldElem) == true
-  @test base_ring(CalciumQQBar) == Union{}
-  @test base_ring(QQBarFieldElem(3)) == Union{}
+  @test base_ring(R) == R
+  @test base_ring(QQBarFieldElem(3)) == R
```
Comment (fingolfin) [inline test/calcium/qqbar-test.jl:11]: Huh, why did the `base_ring` change?

File: test/calcium/qqbar-test.jl [hunk 11]
```diff
@@ -1,14 +1,14 @@
 @testset "QQBarFieldElem.constructors" begin
-  R = CalciumQQBar
+  R = algebraic_closure(QQ)
 
-  @test R == QQBar
+  @test R == Nemo.QQBar
```
Comment (fingolfin) [inline test/calcium/qqbar-test.jl:5]: We don't need this test anymore, do we?
```suggestion
```

File: test/calcium/qqbar-test.jl [hunk 12]
```diff
@@ -41,38 +41,39 @@
...
-
-  @test iszero(R(0))
-  @test isone(R(1))
-  @test is_rational(R(1))
-  @test isreal(R(1))
-  @test degree(R(1)) == 1
-
-  u = sqrt(R(2))
-  i = sqrt(R(-1))
+  QQBar = algebraic_closure(QQ)
```
Comment (fingolfin) [inline test/calcium/qqbar-test.jl:57]: Why not 
```suggestion
  R = algebraic_closure(QQ)
```
and then have the rest of this test stay unchanged?

**Review by fingolfin — APPROVED** [review]
I've tweaked this PR a bit and now I am happy with it in its current state, but now @lgoettgens should verify he is, too ;-)

File: src/calcium/CalciumTypes.jl [hunk 13]
```diff
@@ -86,8 +86,6 @@ Root 0.866025 of 4x^2 - 3
 struct QQBarField <: Field
 end
 
-const QQBar = QQBarField()
-
 @doc qq_field_doc
```
Comment (lgoettgens) [inline src/calcium/CalciumTypes.jl:89]: Off-topic, but this is wrong. It should attach `qqbar_field_doc` instead

**Comment by lgoettgens** [comment]
Thanks @fingolfin for bringing this pr to this form. I am happy to have it merged
